### DISCUSSION TITLE
Segfault when manually starting with ./Dynaframe from ssh console

### DISCUSSION THREAD

**Discussion by Spinner65** [Geektoolkit/Dynaframe3#88] [Q&A]
I just installed 2.16 on Raspberry Pi 4 B rev 1.2 with Buster

If I exit Dynaframe via the webui, and eventually try start it again manually I get a segfault.
(Dynaframe runs fine automatically when you boot)

```
cd /home/pi/Dynaframe
./Dynaframe
```

**Comment by qwksilver** [comment]
i tend to prefer rebooting (sudo reboot) or runing ./run.sh from the Dynaframe directory to restart, it keeps all the logging running as well with either of those methods.

running in terminal or shh(i think)with ./Dynaframe will send the log output live to the terminal/ssh window.

the seg fault was a problem with the first realease of 2.16, if you haven't reinstalled since day 1, you might try again now and see if that cures the issue, let us know.

**Reply by Spinner65** [reply]
both ./run.sh and ./Dynaframe cause a segfault.  
Running the current version of 2.16. I upgraded with the upgrade script just to be sure
Rebooting works fine.
Unfortunately it segfaults before anything gets written to the Dynaframe logs. Just a blank file.

official docs state:
"To restart on a raspberry pi, you can launch it again using ./Dynaframe"

Should I add info about using `./run.sh` to my FAQ thread ?

**Reply by Geektoolkit** [reply]
Yeah, please note it in the FAQ thread.  I'm not sure why it'd crash, it's really odd. Having it work after a reboot rules out a ton of stuff.  Having it crash so early rules out even more (bad image file, large image file, etc.).  Are you using sudo to run it? if so, please don't, that messes it up (the pictures folder doesn't exist for sudo so it doesn't know where to get images).  Also do you have any advanced configuration things like a ton of image paths or odd image paths, anything with special characters?  Do you have any other software running that may be using ports?  This is definitely odd, I'd expect something to get logged at a minimum.

**Comment by msmcknight** [comment]
I had the same problem.  I solved it by exporting my display id and xauthority file location before running Dynaframe.
For example:
$ export DISPLAY=:0.0
$ export XAUTHORITY=~/.Xauthority
$ ./Dynaframe

Your display id may vary, but if you only have one display connected, then it's most likely going to be display :0.0

Good luck!

**Comment by NoMoreQueuing** [comment]
Dynaframe 2.23 (calls itself 2.13)
Raspberry Pi Zero W with Buster
DISPLAY and XAUTHORITY both set
Running on VNC Viewer with Direct Capture Mode
Seg fault on startup, no extra details in logs 
Any ideas?

**Reply by Geektoolkit** [reply]
Yeah Pi zero isn't supported, it doesn't have enough horsepower to pull this off. If youre looking for a slideshow on a pizero you can likely try feh or ibm (sudo apt get should find them both).  This requires a pi3 minimum, and ideally a pi4 for the blurring/transitions and media effects it uses

**Reply by NoMoreQueuing** [reply]
Thank you.  
Dynaframe(1) does most of what I need but without the fade but since I wanted the Pi to hang off the TV USB port, I'll stick with the lower form factor.

**Reply by Geektoolkit** [reply]
Perfect! That's why I kept version 1 around, it's python and very lightweight so it should work on most anything.  Version 2 works on 3 and up, and I have a feeling that as I head to version 3 it's going to be pi4's and up.  I'd love to see more small and powerful form factors arrive...for instance a pi zero vnext that has the power and memory of a pi3 would be amazing.  The Pi3A is cool but it has half the memory, so even with that one it's easy to segfault if you don't watch the image file sizes :/  I'm glad that V1 is at least working for you!

**Comment by Archonw** [comment]
It looks like  i have  the same issue on my raspberry pi 3.
Version 2.19 was the last that started up. Now with 2.23 it doesn't start. When i try it over ssh i get the message segfault.

Here is my log:
Unhandled exception. System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
   at Dynaframe3.PlayListEngine.GetPlayListItems()
   at Dynaframe3.PlayListEngine.GoToNext()
   at Dynaframe3.MainWindow.Timer_Tick(Object sender, EventArgs e)
   at Dynaframe3.MainWindow.InitializeComponent()
   at Dynaframe3.MainWindow..ctor()
   at Dynaframe3.App.OnFrameworkInitializationCompleted()
   at Avalonia.Controls.AppBuilderBase`1.Setup() in /_/src/Avalonia.Controls/AppBuilderBase.cs:line 312
   at Avalonia.Controls.AppBuilderBase`1.SetupWithLifetime(IApplicationLifetime lifetime) in /_/src/Avalonia.Controls/AppBuilderBase.cs:line 179   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime[T](T builder, String[] args, ShutdownMode shutdownMode) in /_/src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs:line 133
   at Dynaframe3.Program.Main(String[] args)

And the output when  i type
./run.sh
./run.sh: Zeile 23:  1080 Speicherzugriffsfehler  ./Dynaframe > /home/pi/Dynaframe/logs/dynaframe-${now}.log 2>&1

**Comment by dankinchen** [comment]
I also have the same problem on my pi 3. I am in VNC viewer. After install the IP scrolls by way to fast for me to read the screen. 

So I did the following;

- pressed escape to terminate the program
-  opened the terminal and typed  ./Dynaframe (did not work)
- found this thread and typed ./Dynaframe/run.sh (and it ran, IP to fast again)
- stuck could not escape or anything, screen is stuck with my first image (no menu's and escape does not work)
- restarted pi by pulling the power and when it booted back up (stuck again at first image)

How do I get it to stop now?

**Reply by dankinchen** [reply]
I restarted it without the keyboard and mouse (this stop the rotations were working) but could not escape.

I found how to bring up the terminal CTRL+ALT+T and then I entered "sudo pkill Dynaframe" that I found in another post. You should add that to the home page.

Also "http://(your_ipaddress):8000" does not work for me. It says this site can't be reach. "ERR_CONNECTION_REFUSED"

**Reply by dankinchen** [reply]
Ok I figured it out. I could not visit the address because Dynaframe was not running, I found another post that helped.

I restarted it using 

- cd Dynaframe
- ./Dynaframe

then I was able to visit the settings page.
